image_transport/test/test_subscriber.cpp

```diff
@@ -22,7 +22,10 @@ TEST_F(TestPublisher, construction_and_destruction) {
   std::function<void(const sensor_msgs::msg::Image::ConstSharedPtr & msg)> fcn =
     [](const auto & msg) {(void)msg;};
 
-  auto sub = image_transport::create_subscription(node_.get(), "camera/image", fcn, "raw");
+  auto qos_profile = rmw_qos_profile_default;
+  auto options = rclcpp::SubscriptionOptionsWithAllocator<std::allocator<void>>();
+  auto sub = image_transport::create_subscription(
+    node_.get(), "camera/image", fcn, "raw", qos_profile, options);
```

Comment:
I suppose this change shouldn't be necessary, but maybe it's worth adding a separate test to try the construction both with and without parameter defaults.